src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -391,9 +391,29 @@ private fun resolveAstNode(
                         resolveTypeDecl(node.typeDecl, resolverState)
                     } ?: resolverState.resolvedEnvironment.typeOf(node.initializer!!),
                 )
+
```

Comment:
As these handling blocks get bigger at some point we might consider extracting them out into their own helpers. The Global.Variable one is probably still small enough, but bear this in mind and feel free to do such extractions as and when you see fit.